### DISCUSSION TITLE
Switching esc with caps lock

### DISCUSSION THREAD

**Discussion by robertorodriguezarch** [JetBrains/ideavim#1011] [Q&A] [answered]
How do I make it so that I can press caps lock instead of ESC to go to normal mode?

**Comment by citizenmatt** [comment, marked as answer]
This has to be handled at the OS level, rather than within IdeaVim, since caps lock is a stateful key that enables/disables caps lock, rather than a simple key press like escape. If you're on Mac, you can use the Keyboard page of System Settings to change Keyboard Shortcuts, select Modifier Keys and change caps lock to map to Escape.

I don't know how to do this in Windows or Linux, but I suspect a web search will be able to suggest a solution.

<img width="718" alt="Screenshot 2024-10-22 at 11 02 00 pm" src="https://github.com/user-attachments/assets/89707242-7165-4d67-9449-ac0a51929683">

**Reply by robertorodriguezarch** [reply]
Yes, that is what I did. Thank you!

**Reply by OptimusCrime** [reply]
First thing that I do when a set up a new Macbook. 

For Windows you can either use [AutoHotKey](https://www.autohotkey.com/), or configured it in the keyboard software (like [Logitech](https://www.logitech.com/en-us/software/options.html) for example).

**Reply by sharpchen** [reply]
> For Windows you can either use [AutoHotKey](https://www.autohotkey.com/)

Or use PowerToys :)

**Reply by lippfi** [reply]
For Linux you can use `setxkbmap -option caps:swapescape`

But the best option is to buy a programmable keyboard. It will work on all Windows, Mac, and Linux without any struggle with settings. And... it will be a good keyboard

**Comment by ayzenquwe** [comment]
I would also recommend to check out [Karabiner](https://karabiner-elements.pqrs.org) if you're on mac. For example, it allows you to remap ctrl+P/N to up/down arrows (only for Idea) which helps to navigate context menus or project/structure panels.

You do that in "Complex Modifications":
```
{
    "description": "Remap ctrl+p/n to arrow up/down in IntelliJ IDEA only",
    "manipulators": [
        {
            "conditions": [
                {
                    "bundle_identifiers": [
                        "^com.jetbrains.intellij"
                    ],
                    "type": "frontmost_application_if"
                }
            ],
            "from": {
                "key_code": "p",
                "modifiers": {
                    "mandatory": [
                        "control"
                    ]
                }
            },
            "to": [
                {
                    "key_code": "up_arrow"
                }
            ],
            "type": "basic"
        },
        {
            "conditions": [
                {
                    "bundle_identifiers": [
                        "^com.jetbrains.intellij"
                    ],
                    "type": "frontmost_application_if"
                }
            ],
            "from": {
                "key_code": "n",
                "modifiers": {
                    "mandatory": [
                        "control"
                    ]
                }
            },
            "to": [
                {
                    "key_code": "down_arrow"
                }
            ],
            "type": "basic"
        }
    ]
}
```
